_posts/2017-01-08-lets-begin-scala-with-gitbucket.md

```diff
@@ -0,0 +1,350 @@
+
+```scala
+def view(title: String){
+  return "<h1>" + escape(title) + "</h1>";
+}
+```
+
+`def` is method in Scala. It's same as `function` keyword in JavaScript. `title: String` is one of arguments of this method. `title` is argument name and `String` is type name. In brief, this method takes an argument named `title` which is `String` type from controller, and use it in the template.
+
+A part that actually uses this argument is `<h1>@title</h1>`. When rewritten in PHP it becomes like this:
+
+```html
+<h1><?php echo escape(title) ?></h1>
+```
+
+In Twirl, values are escaped automatically for XSS protection.
+
+It's also available functio call, if or for expression after `@`:
+
+```html
+@for(order <- orders) {
+  <li>@order.getTitle()</li>
+}
+```
+
+See [the template engine](https://www.playframework.com/documentation/2.5.x/ScalaTemplates) section in the Play documentation to learn more about Twirl. Perhaps, you may find a notice in that documentation:
+
+> Declaring code block this way in a template can be sometime useful but keep in mind that a template is not the best place to write complex logic. It is often better to externalize these kind of code in a Scala class (that you can store under the views/ package as well if you want).
+
+Yes. We shouldn't write complex logic in view template. Of course, view templates in GitBucket don't contain much complex logic (Maybe... I believe so).
+
+GitBucket UI is based on [Bootstrap](http://getbootstrap.com/) and using jQuery in JavaScript.
+
+In this section, you learned Twirl and basis of GitBucket UI. Now you can change UI, replace a logo or change small behavior of GitBucket.
+
+## Controller (Scalatra web framework)
+
+GitBucket uses [Scalatra](http://www.scalatra.org/) web framework. Scalatra is a web framework for Scala inspired by Ruby's Sinatra. In Scala, Play is a major web framework. Play is based on non-blocking I/O, but since Scalatra is based on Java servlet, it can make use of existing servlet based resources. For example, GitBucket is contains JGitServlet which is provided by JGit.
+
+You can write servlet very easy as following:
+
+```scala
+class MyController extends ScalatraServlet with ScalateSupport {
+  get("/hoge") {
+    "Hello, world!"
+  }
+}
+```
+
+Try to re-write more Java like (but this code does not work):
+
+```scala
+class MyController extends ScalatraServlet, ScalateSupport {
+  // コンストラクタ
+  public MyController(){
+    Route route1 = new Route("GET", "/hoge")
+    route1.setAction(new Action(){
+      pulbic Object apply(){
+        return "Hello, world!";
+      }
+    });
+    this.addRoute(route1);
+  }
+}
+```
+
+If you have experience of web application development, you can understand what this code is doing. In Scala, we can write constructor statements in the class definition directory, so DSL such as above example makes easy.
```

Comment:
Not sure I fully understand what you mean in the second sentence (starting from _In Scala_), can you rephrase?